ray_write_shapefiles.py

```diff
@@ -0,0 +1,155 @@
+from datetime import datetime
+from io import BytesIO 
+import os
+import shutil
+from typing import Any, Dict
+
+from osgeo import gdal, osr
+from shapely.geometry import Polygon, Point
+import shapefile
+
+import gdal_virtual_file_path as gdal_vfp
+from mpl_config import MPL_Config
+
+
+def delete_and_create_dir(dir: str):
+    try:
+        shutil.rmtree(dir)
+    except Exception as e:
+        print(f"Error deleting directory {dir}: {e}")
+
+    try:
+        os.makedirs(dir, exist_ok=True)
+        print(f"Created directory {dir}")
+    except Exception as e:
+        print(f"Error creating directory {dir}: {e}")
+
+
+class WriteShapefiles:
+    def __init__(
+        self,
+        config: MPL_Config
+    ):
+        self.config = config
+        self.current_timestamp_str = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+        if self.config.GCP_FILESYSTEM is None:
+            delete_and_create_dir(self.config.RAY_OUTPUT_SHAPEFILES_DIR)
```

Comment:
This logic makes it so that concurrently running actors might accidentally delete each other's directories (ran into this my self). We should try to come up with a less destructive way of doing this or one that plays nicer with concurrency.

Reply:
yeah your'e right - this is problematic - thanks for point it out :)
I moved the create dir to main since we only need to check if the dir exists and create it once (no need to check for each actor). I don't delete the output dir beacuse the output files have unique dates for each run. I think it's nicer to have the output from multiple runs to be able to compare. 

There is still work to be done for creating gcp dirs if they don't exist. I left a comment and a todo for that. The directory management could be cleaned up in future work